src/test-provider/types.ts

```diff
@@ -0,0 +1,94 @@
+import * as vscode from "vscode";
+import { Location } from "vscode-languageclient/node";
+
+/**
+ * Additional information about tests which is stored in map and retrieved when test is scheduled to run.
+ */
+export interface TestItemMetadata {
+  kind: "project" | "package" | "class";
+  targetName: string;
+  targetUri: string;
+}
+
+/**
+ * Information about test classes which is returned by Metals Language Server
+ */
+export interface SuiteDiscovery {
+  targetName: string;
+  targetUri: string;
+  discovered: TestDiscoveryResult[];
+}
+
+export type TestDiscoveryResult = SuiteDiscovery | PackageDiscovery;
+
+export interface SuiteDiscovery {
+  kind: "suite";
+  className: string;
+  fullyQualifiedName: string;
+  location: Location;
+}
+
+export interface PackageDiscovery {
+  kind: "package";
+  prefix: string;
+  children: TestDiscoveryResult[];
+}
+
+/**
+ * Subset of https://microsoft.github.io/debug-adapter-protocol/specification#Events_Output
+ * For our purposes we only care about testResult event which is used to pass information about test results from DAP server to the client.
+ **/
+export type DapEvent =
+  | {
+      event: "testResult";
+      body: {
+        category: "testResult";
+        data: TestSuiteSummary;
+      };
+    }
+  | {
+      event: "output" | "exited" | "terminated";
+      body: {
+        category: "console" | "stdout" | "stderr" | string;
+        output: string;
+      };
+    };
+
+export interface TestSuiteSummary {
+  suiteName: string;
+  duration: number;
+  tests: SingleTestResult[];
+}
+
+export type SingleTestResult = Passed | Failed | Skipped;
+
+interface Skipped {
+  kind: "skipped";
+  testName: string;
+}
+
+interface Passed {
+  kind: "passed";
+  testName: string;
+  duration: number;
+}
+
+export interface Failed {
+  kind: "failed";
+  testName: string;
+  duration: number;
+  error: string;
+}
+
+/**
+ * Describes actions from vscode.TestRun
+ */
+export interface TestRunActions {
```

Comment:
Why do we need an additional interface here?

Reply:
It'll allow testing of `analyzeTestRun` function without the need of `vscode.TestRun` instance. I want to add them later when test explorer will have single test granularity. 

I could remove it until it'll be really needed, but I was too lazy :P